rmgpy/cantherm/explorer.py

```diff
@@ -0,0 +1,236 @@
+        
+        if outputFile:
+            reactionModel.pressureDependence.outputFile = os.path.dirname(outputFile)
+        
+        kineticsDatabase = getDB('kinetics')
+        thermoDatabase = getDB('thermo')
+        
+        thermoDatabase.libraries['thermojobs'] = thermoLibrary
+        thermoDatabase.libraryOrder.insert(0,'thermojobs')
+        
+        kineticsDatabase.libraries['kineticsjobs'] = kineticsLibrary
+        kineticsDatabase.libraryOrder.insert(0,('kineticsjobs','Reaction Library'))
+        
+        reactionModel.addSeedMechanismToCore('kineticsjobs')
+        
+        jobRxns = [rxn for rxn in reactionModel.core.reactions]
+        
+        self.jobRxns = jobRxns
+        
+        for lib in kineticsDatabase.libraryOrder:
+            if lib[0] != 'kineticsjobs':
+                reactionModel.addReactionLibraryToEdge(lib[0])
+        
+        if outputFile is not None:
+            if not os.path.exists(os.path.join(reactionModel.pressureDependence.outputFile,'pdep')):
+                os.mkdir(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+            else:
+                shutil.rmtree(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+                os.mkdir(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+            
+        #get the molecular formula for the network
+        
+        mmol = None
+        for spc in self.source:
+            if mmol:
+                mmol.merge(spc.molecule[0])
+            else:
+                mmol = spc.molecule[0]
+
+        form = mmol.getFormula()
+        
+        for spec in self.pdepjob.network.bathGas.keys():
+            nspec,isNew = reactionModel.makeNewSpecies(spec,reactive=False)
+            flags = np.array([s.molecule[0].getFormula()==form for s in reactionModel.core.species])
+            reactionModel.enlarge(nspec,reactEdge=False,unimolecularReact=flags,
+                    bimolecularReact=np.zeros((len(reactionModel.core.species),len(reactionModel.core.species))))
+        
+        for spc in reactionModel.core.species:
+            for i,item in enumerate(self.source):
+                if spc.isIsomorphic(item):
+                    self.source[i] = spc
+        
+        
+        #react initial species
+        flags = np.array([s.molecule[0].getFormula()==form for s in reactionModel.core.species])
+        logging.info('flags: {0}'.format(flags))
+        reactionModel.enlarge(reactEdge=True,unimolecularReact=flags,
+                      bimolecularReact=np.zeros((len(reactionModel.core.species),len(reactionModel.core.species))))
+        
+        #find the network we're interested in
+        for nwk in reactionModel.networkList:
+            if set(nwk.source) == set(self.source):
+                self.source = nwk.source
+                network = nwk
+                break
+        else:
+            raise ValueError('did not generate a network with the requested source')
```

Comment:
Use `NetworkError` instead of `ValueError`

Reply:
My issue with that is that NetworkError tends to imply there's something wrong with the behavior of the network object, while the actual problem is with which networks RMG decided to generate.